code/game/machinery/computer/fax_responder_spy.dm

```diff
@@ -0,0 +1,329 @@
+	RegisterSignal(spy_eye, COMSIG_PARENT_QDELETING, PROC_REF(remove_current_operator))
+
+/obj/structure/machinery/computer/spy_camera/proc/remove_current_operator()
+	SIGNAL_HANDLER
+	if(!operator) return
+
+	if(spy_eye)
+		last_location = spy_eye.loc
+		if(spy_eye.gc_destroyed)
+			spy_eye = null
+		else
+			QDEL_NULL(spy_eye)
+
+	UnregisterSignal(operator, list(
+		COMSIG_PARENT_QDELETING,
+		COMSIG_MOVABLE_PRE_MOVE,
+		COMSIG_MOB_POST_CLICK
+	))
+	operator.update_sight()
+	var/datum/mob_hud/spy_hud = GLOB.huds[MOB_HUD_SPYCAMS]
+	spy_hud.remove_hud_from(operator, src)
+	operator = null
+
+/obj/structure/machinery/computer/spy_camera/attack_hand(mob/living/carbon/human/pos_operator)
+	if(..())
+		return
+
+	if(!istype(pos_operator))
+		return
+
+	if(locked || (!(pos_operator.faction == FACTION_FAX) && !(spy_faction in pos_operator.faction_group)))
+		to_chat(pos_operator, SPAN_WARNING("The remote camera system is locked out!"))
+		return FALSE
+
+	if(operator && operator.stat == CONSCIOUS)
+		to_chat(pos_operator, SPAN_WARNING("Someone is already using this computer!"))
+		return
+
+	if(tgui_alert(pos_operator, "Change the camera focus?", "Spycam Computer", list("Yes", "No")) == "Yes")
+		var/obj/effect/landmark/spycam_start/start_point = tgui_input_list(pos_operator, "Where do you want to focus the camera?", "Camera Focus", GLOB.spycam_starts)
+		if(!start_point)
+			return
+		last_location = start_point.loc
+
+	set_operator(pos_operator)
+
+
+/obj/effect/landmark/spycam_start
+	name = "Spycam Landmark"
+	icon_state = "spycam"
+
+/obj/effect/landmark/spycam_start/Initialize()
+	. = ..()
+	name = "Spycam [get_area_name(src, TRUE)]"
+	GLOB.spycam_starts += src
+
+/obj/effect/landmark/spycam_start/Destroy()
+	GLOB.spycam_starts -= src
+	return ..()
+
+/mob/hologram/spy_camera
+	name = "Spy Camera"
+	motion_sensed = FALSE
+	icon_state = "spycam"
+
+	color = "#10948d"
+
+	hud_possible = list(SPYCAM_HUD)
+	hears_speech = TRUE
+
+	var/mob/living/carbon/is_watching
+
+	var/spy_range = 5
+	var/spy_faction = FACTION_NEUTRAL
+
+	var/list/temporary_list = list()
+	var/list/temporary_list_2 = list()
+
+	///Whether or not the camera is on cooldown for a warning message it can't move to a certain tile, locked to one message every 3 seconds.
+	var/move_warn = FALSE
+
+
+/mob/hologram/spy_camera/Initialize(mapload, mob/living/carbon/spy_operator, obj/structure/machinery/computer/spy_camera/console)
+	if(!console || !spy_operator)
+		return INITIALIZE_HINT_QDEL
+
+	if(!istype(console))
+		stack_trace("Tried to initialize a /mob/hologram/spy_camera on type ([console.type])")
+		return INITIALIZE_HINT_QDEL
+
+	spy_range = console.spy_range
+	spy_faction = console.spy_faction
+	faction = spy_faction
+
+	switch(spy_faction)
+		if(FACTION_MARINE, FACTION_MARSHAL)
+			color = "#0947bb"
+		if(FACTION_CLF)
+			color = "#717fbd"
+		if(FACTION_UPP)
+			color = "#0f3d11"
+		if(FACTION_TWE)
+			color = "#b350c0"
+		if(FACTION_WY)
+			color = "#b6b6b6"
+
+	. = ..()
```

Comment:
just curious why this hanging out down here of all places-- don't think it would critically break things but its unorthodox.